src/Azure.Functions.Cli/Kubernetes/KEDA/V2/KedaV2Resource.cs

```diff
@@ -41,20 +41,56 @@ public class KedaV2Resource : KedaResourceBase
                         .Where(b => b?["type"] != null)
                         .Where(b => b["type"].ToString().IndexOf("Trigger", StringComparison.OrdinalIgnoreCase) != -1)
                         .Where(b => b["type"].ToString().IndexOf("httpTrigger", StringComparison.OrdinalIgnoreCase) == -1)
-                        .Select(t => new ScaledObjectTriggerV1Alpha1
-                        {
-                            Type = GetKedaTriggerType(t["type"]?.ToString()),
-                            Metadata = PopulateMetadataDictionary(t)
-                        })
+                        .GroupBy(b => IsDurable(b)) // Multiple durable triggers map to a single scaler
+                        .SelectMany(group => group.Key ? GetDurableScalar(triggers.HostJson) : group.Select(GetStandardScalar))
                 }
             };
         }
 
-        private const string ConnectionField = "connection";
-        private const string ConnectionFromEnvField = "connectionFromEnv";
+        private static bool IsDurable(JToken trigger) => 
+            trigger["type"].ToString().Equals("orchestrationTrigger", StringComparison.OrdinalIgnoreCase) ||
+            trigger["type"].ToString().Equals("activityTrigger", StringComparison.OrdinalIgnoreCase) ||
+            trigger["type"].ToString().Equals("entityTrigger", StringComparison.OrdinalIgnoreCase);
+
+        private static IEnumerable<ScaledObjectTriggerV1Alpha1> GetDurableScalar(JObject hostJson)
+        {
+            // Reference: https://docs.microsoft.com/azure/azure-functions/durable/durable-functions-bindings#durable-functions-2-0-host-json
+            JObject storageProviderConfig = hostJson.SelectToken("extensions.durableTask.storageProvider") as JObject;
+            string storageType = storageProviderConfig?["type"]?.ToString();
+
+            // Custom storage types are supported starting in Durable Functions v2.4.2
+            if (string.Equals(storageType, "MicrosoftSQL", StringComparison.OrdinalIgnoreCase))
+            {
+                yield return new ScaledObjectTriggerV1Alpha1
+                {
+                    // MSSQL scaler reference: https://keda.sh/docs/2.2/scalers/mssql/
+                    Type = "mssql",
+                    Metadata = new Dictionary<string, string>
+                    {
+                        ["connectionStringFromEnv"] = storageProviderConfig?["connectionStringName"]?.ToString(),
```

Comment:
do you need to add other metadata for the scaler, `query`, `targetValue`?

Reply:
Yes. I left it out because none of the other Functions triggers actually included this info.  Is that because they have default values if not specified? I can add this.

Reply:
correct. I think the `query` here is analogous to `queueName` for example. We don't set "targetValue" by default since it's not a concept in Azure/Azure Functions, but Keda has always a default for that.